modules/couchbase/src/main/java/org/testcontainers/containers/CouchbaseContainer.java

```diff
@@ -0,0 +1,284 @@
+            }
+            String setupServiceContent = "services=" + URLEncoder.encode(servicePayloadBuilder.toString(), "UTF-8");
+
+            String webSettingsURL = "/settings/web";
+            String webSettingsContent = "username=" + URLEncoder.encode(clusterUsername, "UTF-8") + "&password=" + URLEncoder.encode(clusterPassword, "UTF-8") + "&port=8091";
+
+            String bucketURL = "/sampleBuckets/install";
+
+            StringBuilder sampleBucketPayloadBuilder = new StringBuilder();
+            sampleBucketPayloadBuilder.append('[');
+            if (travelSample) {
+                sampleBucketPayloadBuilder.append("\"travel-sample\",");
+            }
+            if (beerSample) {
+                sampleBucketPayloadBuilder.append("\"beer-sample\",");
+            }
+            if (gamesIMSample) {
+                sampleBucketPayloadBuilder.append("\"gamesim-sample\",");
+            }
+            sampleBucketPayloadBuilder.append(']');
+
+            callCouchbaseRestAPI(poolURL, poolPayload);
+            callCouchbaseRestAPI(setupServicesURL, setupServiceContent);
+            callCouchbaseRestAPI(webSettingsURL, webSettingsContent);
+            callCouchbaseRestAPI(bucketURL, sampleBucketPayloadBuilder.toString());
+
+            CouchbaseWaitStrategy s = new CouchbaseWaitStrategy();
+            s.withBasicCredentials(clusterUsername, clusterPassword);
+            s.waitUntilReady(this);
+            callCouchbaseRestAPI("/settings/indexes", "indexerThreads=0&logLevel=info&maxRollbackPoints=5&storageMode=memory_optimized");
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void createBucket(BucketSettings bucketSetting, Boolean createIndex){
+        BucketSettings bucketSettings = getCouchbaseCluster().clusterManager(clusterUsername, clusterPassword).insertBucket(bucketSetting);
+        // allow some time for the query service to come up
+        try {
+            Thread.sleep(5000);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+        if (createIndex) {
+            getCouchbaseCluster().openBucket().query(Index.createPrimaryIndex().on(bucketSetting.name()));
+        }
+    }
+
+    public void callCouchbaseRestAPI(String url, String payload) throws IOException {
+        String fullUrl = urlBase + url;
+        HttpURLConnection httpConnection = (HttpURLConnection) ((new URL(fullUrl).openConnection()));
+        httpConnection.setDoOutput(true);
+        httpConnection.setRequestMethod("POST");
+        httpConnection.setRequestProperty("Content-Type",
+                "application/x-www-form-urlencoded");
+        String encoded = Base64.encode((clusterUsername + ":" + clusterPassword).getBytes("UTF-8"));
+        httpConnection.setRequestProperty("Authorization", "Basic " + encoded);
+        DataOutputStream out = new DataOutputStream(httpConnection.getOutputStream());
```

Comment:
add `@lombok.Cleanup DataOutputStream out` to automatically close the stream (no need to call `out.close()`, which should be called in `try {} finally {}`)